spark-plugins/src/main/java/co/cask/hydrator/plugin/batch/spark/NaiveBayesTrainer.java

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.spark;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.SparkPluginContext;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.mapreduce.lib.input.TextInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.TextOutputFormat;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.mllib.classification.NaiveBayes;
+import org.apache.spark.mllib.classification.NaiveBayesModel;
+import org.apache.spark.mllib.feature.HashingTF;
+import org.apache.spark.mllib.regression.LabeledPoint;
+
+/**
+ * Spark Sink plugin that trains a model based upon whether messages are spam or not, and then classifies messages.
+ */
+@Plugin(type = SparkSink.PLUGIN_TYPE)
+@Name(NaiveBayesTrainer.PLUGIN_NAME)
+@Description("Trains a model based upon whether messages are spam or not.")
+public final class NaiveBayesTrainer extends SparkSink<StructuredRecord> {
+  public static final String PLUGIN_NAME = "NaiveBayesTrainer";
+
+  private final Config config;
+
+  /**
+   * Configuration for the NaiveBayesTrainer.
+   */
+  public static class Config extends PluginConfig {
+
+    @Description("FileSet to use to load the model from.")
+    private final String fileSetName;
+
+    @Description("Path of the FileSet to load the model from.")
+    private final String path;
+
+    @Description("A space-separated sequence of words, which to use for classification.")
+    private final String fieldToClassify;
+
+    @Description("The field from which to get the prediction. It must be of type double.")
+    private final String predictionField;
+
+    public Config(String fileSetName, String path, String fieldToClassify, String predictionField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.fieldToClassify = fieldToClassify;
+      this.predictionField = predictionField;
+    }
+  }
+
+  public NaiveBayesTrainer(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class, FileSetProperties.builder()
```

Comment:
should do some validation too.  For example, if the input schema is non-null, make sure the fieldToClassify exists and is of type String, and the predictionField exists and is of type double